bgpd/bgp_nht.c

```diff
@@ -297,16 +296,21 @@ void bgp_delete_connected_nexthop(afi_t afi, struct peer *peer)
 
 	rn = bgp_node_lookup(
 		peer->bgp->nexthop_cache_table[family2afi(p.family)], &p);
-	if (!rn || !rn->info) {
+	if (!rn) {
 		if (BGP_DEBUG(nht, NHT))
 			zlog_debug("Cannot find connected NHT node for peer %s",
 				   peer->host);
-		if (rn)
-			bgp_unlock_node(rn);
 		return;
 	}
 
-	bnc = rn->info;
+	bnc = bgp_nexthop_from_node(rn);
+	if (!bnc) {
+		if (BGP_DEBUG(nht, NHT))
+			zlog_debug("Cannot find connected NHT node for peer %s",
```

Comment:
Change to "node info" to allow debugger to understand different cases.  Alternatively, keep combined code structure...